src/ess/polarization/base.py

```diff
@@ -353,67 +342,132 @@ def he3_opacity_from_cell_params(
         * pressure
         * length
     )
-    return He3OpacityFunction[Cell](OpacityFunction(opacity0))
+    return He3OpacityFunction[Cell](opacity0)
 
 
 def he3_opacity_from_beam_data(
     transmission_empty_glass: He3TransmissionEmptyGlass[Cell],
     direct_beam: DirectBeamNoCell,
     direct_beam_cell: He3DirectBeam[Cell, Depolarized],
+    pressure: He3CellPressure[Cell],
+    length: He3CellLength[Cell],
+    temperature: He3CellTemperature[Cell],
 ) -> He3OpacityFunction[Cell]:
     """
     Opacity function for a given cell, based on direct beam data.
 
     Note that this can alternatively be defined via cell parameters, see
     :py:func:`he3_opacity_from_cell_params`.
     """
+    opacity0 = he3_opacity_from_cell_params(pressure, length, temperature).opacity0
 
     def intensity(wavelength: sc.Variable, opacity0: sc.Variable) -> sc.Variable:
-        opacity = OpacityFunction(opacity0)
+        opacity = He3OpacityFunction[Cell](opacity0)
         return transmission_empty_glass * sc.exp(-opacity(wavelength))
 
     popt, _ = sc.curve_fit(
         ['wavelength'],
         intensity,
         direct_beam_cell / direct_beam,
-        # TODO We could use opacity0 from cell parameters as initial guess.
-        p0={'opacity0': sc.scalar(1.0, unit='1/nm')},
+        p0={'opacity0': opacity0},
     )
-    return He3OpacityFunction[Cell](OpacityFunction(popt['opacity0'].data))
+    return He3OpacityFunction[Cell](popt['opacity0'].data)
+
+
+class He3PolarizationFunction(Generic[Cell]):
+    """Time-dependent polarization function for a given cell."""
+
+    def __init__(self, C: sc.Variable, T1: sc.Variable):
+        self._C = C
+        self._T1 = T1
+
+    @property
+    def C(self) -> sc.Variable:
+        return self._C
+
+    @property
+    def T1(self) -> sc.Variable:
+        return self._T1
 
+    def __call__(self, time: sc.Variable) -> sc.Variable:
+        return self.C * sc.exp(-time / self.T1)
 
-def he3_polarization(
+
+@dataclass
+class He3TransmissionFunction(Generic[Cell]):
+    """Wavelength- and time-dependent transmission for a given cell."""
+
+    opacity_function: He3OpacityFunction[Cell]
+    polarization_function: He3PolarizationFunction[Cell]
+    transmission_empty_glass: He3TransmissionEmptyGlass[Cell]
+
+    def __call__(
+        self,
+        *,
+        time: sc.Variable,
+        wavelength: sc.Variable,
+        plus_minus: Literal['plus', 'minus'],
+    ) -> sc.Variable:
+        opacity = self.opacity_function(wavelength)
+        polarization = self.polarization_function(time)
+        if plus_minus == 'plus':
+            polarization -= 1.0
+        else:
+            polarization += 1.0
+        return self.transmission_empty_glass * sc.exp(opacity * polarization)
```

Comment:
In [the description](https://github.com/scipp/esspolarization/blob/main/docs/user-guide/sans-polarization-analysis-methodology.ipynb) (section "Spin-dependent transmission values") it looks like `plus_minus = 'plus'` implies that the expression in `exp()` should be $-O \lambda (1 + P)$ but in this function I think we get $O (P - 1)$.

Seems to be two things different:
1. ~$\lambda$ does not occur in the final expression, is that because the notation in the description is different, and the wavelength is actually an argument to the opacity like it is in this function?~ Yes
2. The sign of $P$ seems to be wrong.

Reply:
Hm, we want for 'plus' : -OP-1, and for 'minus': +OP-O.

Reply:
Good catch, should be fixed now, please have another look.

@astellhorn No, this is not used during the fit.